python/mujinplanningclient/binpickingplanningclient.py

```diff
@@ -46,641 +75,2953 @@ def __init__(self, regionname=None, **kwargs):
             ignoredArgs: Additional keyword args are not used, but allowed for easy initialization from a dictionary
         """
         self.regionname = regionname
-        super(BinpickingPlanningClient, self).__init__(tasktype=self.tasktype, **kwargs)
+        super(BinpickingPlanningClient, self).__init__(
+            robotname=robotname,
+            robotspeed=robotspeed,
+            robotaccelmult=robotaccelmult,
+            envclearance=envclearance,
+            robotBridgeConnectionInfo=robotBridgeConnectionInfo,
+            targetname=targetname,
+            toolname=toolname,
+            taskzmqport=taskzmqport,
+            taskheartbeatport=taskheartbeatport,
+            taskheartbeattimeout=taskheartbeattimeout,
+            tasktype=tasktype,
+            ctx=ctx,
+            slaverequestid=slaverequestid,
+            controllerip=controllerip,
+            controllerurl=controllerurl,
+            controllerusername=controllerusername,
+            controllerpassword=controllerpassword,
+            scenepk=scenepk,
+            callerid=callerid
+        )
 
 
     #
-    # Commands
+    # Commands (generated from the spec)
     #
 
-
     def PickAndPlace(
         self,
-        goaltype=_deprecated,
-        goals=_deprecated,
-        targetnamepattern=None,
-        approachoffset=30,
-        departoffsetdir=[0, 0, 50],
-        destdepartoffsetdir=[0, 0, 30],
-        deletetarget=0,
-        debuglevel=4,
-        movetodestination=1,
-        freeinc=[0.08],
-        worksteplength=None,
-        densowavearmgroup=5,
-        regionname=None,
-        cameranames=None,
-        envclearance=None,
-        toolname=None,
-        robotspeed=None,
-        timeout=1000,
-        **kwargs
+        goaltype=_deprecated,  # type: Optional[str]
+        goals=_deprecated,  # type: Optional[str]
+        targetnamepattern=None,  # type: str
+        approachoffset=30,  # type: float
+        departoffsetdir=[0, 0, 50],  # type: tuple[float, float, float]
```

Comment:
I know we used to do this in the original client, so I'm OK with leaving this as is for now, but we shouldn't generate code with mutable defaults. It's a footgun. Ideally, mutable parameters with a default would do:

```python
def fn(
    departoffsetdir=None,  # type: Optional[tuple[float, float, float]]
    # [...]
):
    taskparameters = {}
    if departoffsetdir is not None:
        taskparameters['departoffsetdir'] = departoffsetdir
    else: 
       taskparameters['departoffsetdir'] = [0, 0, 50]
```

Reply:
Filed an issue internally.